src/tools/compiletest/src/header.rs

```diff
@@ -44,6 +44,8 @@ impl EarlyProps {
             props.ignore =
                 props.ignore ||
                 config.parse_cfg_name_directive(ln, "ignore") ||
+                (config.parse_cfg_prefix(ln, "only") &&
+                !config.parse_cfg_name_directive(ln, "only")) ||
```

Comment:
Why do we need to call `parse_cfg_prefix` here but not for ignore? What makes only special?

Reply:
This checks is there any //only-<> specification present and if so, does it matches with current platform.

1) If there exists a //only-<> and the platform does not matches, this should be skipped
2) If there exists a //only-<> and the platform does match, this should not be skipped

For ignore, we don't care if any other //ignore-<> exists. Like platform 'bar' won't care of //ignore-<foo> but it should care if //only-<foo> exists.

Reply:
Ah, that makes sense. Thanks! Can we put that into a (short) comment in the code perhaps?

Reply:
Sure that will be helpful. I will add that. I am also not sure what should interactions between //only-<> and //ignore-<> should result into. There can be some interesting cases there.